src/foam/i18n/LocaleTranslationService.js

```diff
@@ -20,8 +29,61 @@ foam.CLASS({
     {
       name: 'getTranslation',
       javaCode: `
-      System.err.println("*********************************** getTranslation");
-      return "foobar";
+        if ( locale == "" || source == "" ) return defaultText;
+
+        DAO localeDAO = (DAO) getX().get("localeDAO");
+
+        // localeVariant - complete culture code ('en-CA', ‘fr-CA’, ‘de_AT’ …etc) 
+        // locale - determines locale type (en, fr, es …etc)        
+        // variant - Locale variation (CA for en-CA, CA for fr-CA, AT for de_AT …etc)
+        String variant = "";
+        String localeVariant = "";
+        Locale localeEntry = null;
+
+        // note: 'locale' passed to the function may be in locale format or in local variant format
+        // so we need to check which format is used and parse it if necessary.
+        boolean hasVariant = locale.contains("-");
+        if ( hasVariant ) {
+          variant = locale.substring(3).toUpperCase();
+          locale = locale.substring(0,2).toLowerCase();
+          localeVariant = locale + "-" + variant;
+
+          // search for locale and variant
+          localeEntry = (Locale) localeDAO.find(
+            AND(
+              EQ(Locale.SOURCE, source),        
+              EQ(Locale.LOCALE, locale),
+              EQ(Locale.VARIANT, variant)
+            )
+          );
+
+          // search for locale variant
+          if ( localeEntry == null ) {
+            localeEntry = (Locale) localeDAO.find(
+              AND(
+                EQ(Locale.SOURCE, source),
+                EQ(Locale.LOCALE_VARIANT, localeVariant)
```

Comment:
search for AND(EQ(LOCALE, locale), EQ(VARIANT, ''))

Reply:
Actually, just delete this whole query, since you're doing the proper one below.